### PR TITLE
Release 0.4.2: Improved classification performance

### PR DESCRIPTION
**Pull request by maxhutch**
What a difference a data structure makes!  Classification trees are now up to 6x faster.

### CODE REVIEW

File: src/test/scala/io/citrine/lolo/PerformanceTest.scala [hunk 1]
```diff
@@ -71,9 +75,11 @@ class PerformanceTest {
     * Test the absolute performance to check for overall regressions
     */
   def testAbsolute(): Unit = {
-    val (nominalTrain, nominalPredict) = timedTest(trainingData, 1024, 32, 1024)
-    assert(nominalTrain < 8.0, s"Expected nominal train to have theta < 8.0 but was ${nominalTrain}")
-    assert(nominalPredict < 7.0, s"Expected nominal transform to have theta < 7.0 but was ${nominalPredict}")
+    // val (nominalTrain, nominalPredict) = timedTest(trainingData, 1024, 32, 1024)
+    val (nominalTrain, nominalPredict) = timedTest(classificationData, 1024, 32, 1024)
+    println(nominalTrain, nominalPredict)
```
Comment (eantono) [inline src/test/scala/io/citrine/lolo/PerformanceTest.scala]: remove println

File: src/test/scala/io/citrine/lolo/PerformanceTest.scala [hunk 2]
```diff
@@ -71,9 +75,11 @@ class PerformanceTest {
...
+    // assert(nominalTrain < 8.0, s"Expected nominal train to have theta < 8.0 but was ${nominalTrain}")
```
Comment (eantono) [inline src/test/scala/io/citrine/lolo/PerformanceTest.scala]: uncomment asserts?

File: src/test/scala/io/citrine/lolo/trees/splits/ClassificationSplitterTest.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import io.citrine.lolo.TestUtils
+import io.citrine.lolo.encoders.CategoricalEncoder
+import io.citrine.theta.Stopwatch
+
+
+/**
+  * Hold some performance profiling routines
+  */
+class ClassificationSplitterTest {
```
Comment (eantono) [inline src/test/scala/io/citrine/lolo/trees/splits/ClassificationSplitterTest.scala]: Is meant to evaluate the performance against some benchmark, or just print the time?

**Comment by maxhutch** [comment]
@eantono the performance on travis isn't stable enough to use those performance tests for CI.  You'll notice they don't have the `@Test` annotation on them.  I'll clean them up a bit and add docs to that effect.

**Comment by eantono** [comment]
@maxhutch, there appears to be an outstanding test failure

**Pull request merged**